ghost/collections/src/Entity.ts

```diff
@@ -0,0 +1,104 @@
+
+type Writeable<T> = {
+    -readonly [K in keyof T]: T[K]
+};
+class EntityImpl<Data> {
+    private _attr: Writeable<Data>
+    protected attr: Writeable<Data>
+    protected fields!: (keyof Data)[]
+    protected writeableFields!: (keyof Data)[]
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    protected validations!: any;
+
+    constructor(data: Data) {
+        this._attr = data;
+        this.attr = new Proxy(this._attr, {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            set(target: any, key: any, value: unknown) {
+                if (target[key] !== value) {
+                    if (Reflect.has(target, 'updatedAt')) {
```

Comment:
Another one to think of is probably a property identifying if the entity has been modified or "touched" since instantiation 🤔 Is see this useful in situations where we check if we need to reset cache when the data has been modified. Wdyt?